app/routes/variantText/variantTexts.js

```diff
@@ -0,0 +1,226 @@
+
+const {
+  getUserProjects,
+  isAdmin,
+  sanitizeHtml,
+  projectAccess,
+} = require('../../libs/helperFunctions');
+const schemaGenerator = require('../../schemas/schemaGenerator');
+const validateAgainstSchema = require('../../libs/validateAgainstSchema');
+const {BASE_EXCLUDE} = require('../../schemas/exclude');
+
+const router = express.Router({mergeParams: true});
+
+// Generate schema's
+const createSchema = schemaGenerator(db.models.variantText, {
+  baseUri: '/create', exclude: [...BASE_EXCLUDE],
+});
+const updateSchema = schemaGenerator(db.models.variantText, {
+  baseUri: '/update', include: ['text'], nothingRequired: true,
+});
+
+const pairs = {
+  project: db.models.project,
+  template: db.models.template,
+};
+
+// for each entry in pairs, assumes the key-named value in
+// req.body is the ident, and gets the id of the corresponding object.
+router.use(async (req, res, next) => {
+  const operations = [];
+
+  for (const [key, value] of Object.entries(pairs)) {
+    // delete user input ids for safety
+    delete req.body[`${key}Id`];
+    if (req.body[key]) {
+      const operation = value.findOne({
+        where: {ident: req.body[key]},
+      }).then((obj) => {
+        if (!obj || !obj.id) {
+          logger.error(`Unable to find ${key} ${req.body[key]}`);
+          // Throw an error object that includes a status code
+          const error = new Error(`Unable to find ${key}`);
+          error.statusCode = HTTP_STATUS.NOT_FOUND;
+          throw error;
+        }
+        req.body[`${key}Id`] = obj.id;
+      });
+
+      operations.push(operation);
+    }
+  }
+
+  try {
+    await Promise.all(operations);
+    next();
+  } catch (error) {
+    logger.error(`Error while trying to find key: ${error.message}`);
+    // Use the status code from the error object, if it exists; otherwise, use 500
+    const statusCode = error.statusCode || HTTP_STATUS.INTERNAL_SERVER_ERROR;
+    res.status(statusCode).json({
+      error: {message: `Error while trying to find key: ${error.message}`},
+    });
+  }
+});
+
+// Middleware for germline reviews
+router.param('variantText', async (req, res, next, ident) => {
+  let result;
+  try {
+    result = await db.models.variantText.findOne({
+      where: {ident},
+      include: [
+        {model: db.models.template.scope('minimal'), as: 'template'},
+        {model: db.models.project.scope('minimal'), as: 'project'},
+      ],
+    });
+  } catch (error) {
+    logger.error(`Error while trying to get variant text ${error}`);
+    return res.status(HTTP_STATUS.BAD_REQUEST).json({
+      error: {message: 'Error while trying to get variant text'},
+    });
+  }
+
+  if (!result) {
+    logger.error('Unable to find variant text');
+    return res.status(HTTP_STATUS.NOT_FOUND).json({
+      error: {message: 'Unable to find variant text'},
+    });
+  }
+
+  const userHasProjectAccess = projectAccess(req.user, {projects: [{ident: result.project.ident}]});
```

Comment:
does this return true if there are no projects?

Reply:
It doesnt, do you want me to default to true if there's no project?

Reply:
Yes please!